packages/recommend-react/src/useFrequentlyBoughtTogether.ts

```diff
@@ -28,23 +31,26 @@ export function useFrequentlyBoughtTogether<TObject>({
   const transformItems = useStableValue(userTransformItems);
   const queryParameters = useStableValue(userQueryParameters);
 
-  useAlgoliaAgent({ recommendClient });
+  useAlgoliaAgent({ recommendClient, initialState });
 
   useEffect(() => {
     setStatus('loading');
-    getFrequentlyBoughtTogether({
-      indexName,
-      maxRecommendations,
-      objectIDs,
-      queryParameters,
-      recommendClient,
-      threshold,
-      transformItems,
-    }).then((response) => {
-      setResult(response);
-      setStatus('idle');
-    });
+    if (!initialState) {
```

Comment:
You need to wrap the whole effect in this condition, otherwise you get on a stuck "loading" state.

Also, this effect should be skipped only on first execution. Otherwise you won't ever refetch recommendations when one of the props change.

(I would advise to write all test scenarios, because SSR support complexifies the flow.)